src/wasm.rs

```diff
@@ -148,9 +149,12 @@ fn handle_module(
 
         let func_index = num_func_imports + def_index.index();
         if flag_check_translation {
-            context
-                .verify(fisa)
-                .map_err(|err| pretty_verifier_error(&context.func, fisa.isa, None, &err))?;
+            let mut errors = VerifierErrors::default();
+            let _ = context.verify(fisa, &mut errors);
+
+            if !errors.0.is_empty() {
```

Comment:
Instead of testing `errors.0.is_empty()` here, it feels cleaner to test whether the return value of `verify` `.is_err()`.

Reply:
Ditto, the `Result` may be `Ok` but a (non-fatal) error *might* have been logged anyway.